File: ruby/emit_logs_topic.rb
#!/usr/bin/env ruby
require 'bunny'

# create connection
conn = Bunny.new
conn.start

# create channel
ch = conn.create_channel

# define a topic exchange
x = ch.topic 'topic_logs'

# get message and arguments
severity = ARGV.shift || 'anonymous.info'
msg = ARGV.empty? ? 'Ruby log message...' : ARGV.join(' ')

# publish log to the exchange
x.publish(msg, routing_key: severity)
puts " [x] Sent from Ruby: #{severity}:#{msg}"

# close a connection
conn.close

File: elixir/emit_logs_topic.exs
# create connection
{:ok, connection} = AMQP.Connection.open
{:ok, channel} = AMQP.Channel.open(connection)

# get message and arguments
{topic, message} =
  System.argv
  |> case do
    [] -> {"anonimous.info", "Elixir message"}
    [message] -> {"anonimous.info", message}
    [topic|words] -> {topic, Enum.join(words, " ")}
  end

# create exchange
AMQP.Exchange.declare(channel, "topic_logs", :topic)

# push a log message
AMQP.Basic.publish(channel, "topic_logs", topic, message)
IO.puts " [x] Published from Elixir '[#{topic}] #{message}'"

AMQP.Connection.close(connection)
